### PR TITLE
[network commissioning] Make EnableNetwork async

### PR DESCRIPTION
**Pull request by erjiaqing**
#### Problem

Current EnableNetwork command handling is a sync command call, however, the enable process can be time-consuming, thus make it async.

 #### Summary of Changes

Run EnableNetwork in mainloop after validate input parameters.

 Fixes #<<<<<FILL ME IN  - issue number(s). If no issue, please create one>>>>>

### CODE REVIEW

**Review by msandstedt — APPROVED** [review]

**Comment by msandstedt** [comment]
Are commands themselves handled outside of the main loop?  If so, what are the thread safety implications of moving this call from the command handler context to the main loop context?  If not, how does this improve things?

Also, is it actually safe to make this asynchronous?  Is there a danger that if commands are received in quick succession, subsequent commands may fail if the rely upon this command already being completed?

**Comment by msandstedt** [comment]
> > Are commands themselves handled outside of the main loop? If so, what are the thread safety implications of moving this call from the command handler context to the main loop context? If not, how does this improve things?
> > Also, is it actually safe to make this asynchronous? Is there a danger that if commands are received in quick succession, subsequent commands may fail if the rely upon this command already being completed? I would think something like this requires a more comprehensive story to ensure all actions on this cluster complete in the correct order. If this command is dispatched into the main loop, shouldn't all handling for this cluster be dispatched into the main loop?
> 
> The idea is to check the value provided by use in handler, and do time consuming matters in mainloop.
> 
> Per network commissioning cluster spec, the client should receive a Response command, like EnableNetworkRespond, to indicate the actual success, currently, this mechanism is missing and should be implemented in #5294 , #5619
> 
> Or maybe we should keep the this wokr synchronous until the above functions are ready?

This answers most of my concerns.  And no, I don't personally think this needs to wait for #5294 and #5619 for this to land.  

However, what about thread safety?  It would seem this usage implies that all of the `ThreadStackManagerImpl.cpp`'s need to be thread safe.  I don't see any locking in these at all.  Am I missing something?

File: src/app/clusters/network-commissioning/network-commissioning.cpp [hunk 1]
```diff
@@ -211,18 +211,21 @@ EmberAfNetworkCommissioningError OnAddWiFiNetworkCommandCallbackInternal(app::Co
 }
 
 namespace {
-CHIP_ERROR DoEnableNetwork(NetworkInfo * network)
+void DoEnableNetwork(intptr_t networkInfoPtr)
```
Comment (mspang) [inline src/app/clusters/network-commissioning/network-commissioning.cpp:214]: Why not just pass the index instead of intptr_t ?

**Review by bzbarsky-apple — COMMENTED** [review]
I'm trying to understand why this actually helps.  Does DoEnableNetwork run on a background thread?  Or still on the same thread as `OnEnableNetworkCommandCallbackInternal` but just async?

If the latter, what is the benefit?

**Comment by msandstedt** [comment]
> I'm trying to understand why this actually helps. Does DoEnableNetwork run on a background thread? Or still on the same thread as `OnEnableNetworkCommandCallbackInternal` but just async?
> 
> If the latter, what is the benefit?

This is exactly my point as well.  Either these things can happen in two separate threads, in which case this solves the stated problem but reveals a problematic lack of thread safety in all ThreadStackManagerImpl.cpp's, or this change doesn't actually help anything.

As far as I can tell, one of these two things has to be true.

**Comment by erjiaqing** [comment]
Close this pr since we are working on new network commissioning impl.

**Pull request closed without merging**